src/csharp/Grpc.Core.Tests/CompressionTest.cs

```diff
@@ -118,5 +119,30 @@ public void WriteOptions_Unary()
 
             await call.ResponseStream.ToListAsync();
         }
+
+        [Test]
+        public void CanReadCompressedMessages()
+        {
+            var compressionMetadata = new Metadata
+            {
+                { new Metadata.Entry("grpc-internal-encoding-request", "gzip") }
```

Comment:
Shouldn't this be encapsulated, like in the `CreateClientCompressionMetadata` method you use in other places?